lib/clients/transports/request.js

```diff
@@ -64,7 +65,8 @@ var proxiedRequestTransport = function proxiedRequestTransport(proxyURL) {
 
 var requestOptionsTransport = function requestOptionsTransport(options) {
   return function _requestOptionsTransport(args, cb) {
-    var requestArgs = defaults(options, getRequestTransportArgs(args));
+    var instanceOptions = cloneDeep(options);
```

Comment:
could this line be outside of the inner `_requestOptionsTransport` function? or in other words, what's the point in cloning the `options` each time a request is fired, as opposed to just cloning it once per transport "constructed"?

Reply:
I think the problem is that the way defaults works is by putting instanceOptions as the base of requestArgs and then layering the args on top of it. This means that after the args are layered on top, instanceOptions has now been overridden with the args specified by the caller.

This is problematic for the next request because then instanceOptions now also includes the args from the previous request. By ensuring that the original options object doesn't mutate and cloning it for each request, we avoid this issue.

Reply:
ah! i was assuming `defaults()` is a pure function. silly lodash developers. 👍